Gems/ScriptCanvas/Code/Include/ScriptCanvas/Core/Datum.cpp

```diff
@@ -2041,7 +2041,22 @@ namespace ScriptCanvas
         }
     }
 
-    void Datum::OnWriteEnd()
+    void Datum::OnSerializeBegin()
+    {
+        if (m_type.GetType() == Data::eType::BehaviorContextObject)
+        {
+            if (BehaviorContextObjectPtr ptr = (*AZStd::any_cast<BehaviorContextObjectPtr>(&m_storage.value)))
+            {
+                ptr->OnSerializeBegin();
+            }
+            else
+            {
+                AZ_Error("ScriptCanvas", false, AZStd::string::format("Datum type (%s) failed to serialized, did not store BehaviorContextObjectPtr properly", m_type.GetAZType().ToString<AZStd::string>().c_str()).c_str());
+            }
+        }
+    }
+
+    void Datum::OnDeserialize()
```

Comment:
Should this block also be guarded with `#if defined(OBJECT_STREAM_EDITOR_ASSET_LOADING_SUPPORT_ENABLED)`?

Reply:
No, this is required for the JSON serialization, too